### PR TITLE
Support metadata table pg_tables,pg_roles in og scenario.

### PR DESCRIPTION
**Pull request by JiekerTime**
Enhance https://github.com/apache/shardingsphere/pull/25741

Liked to issue #25468 #24662. maybe.

Features:

image

image

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by JiekerTime** [comment]
@sandynz

**Review by tuichenchuxin — APPROVED** [review]

File: proxy/backend/type/opengauss/src/main/java/org/apache/shardingsphere/proxy/backend/opengauss/handler/admin/OpenGaussSystemCatalogAdminQueryExecutor.java [hunk 1]
```diff
@@ -88,11 +97,24 @@ private void prepareCalciteConnection(final Connection connection) throws SQLExc
...
-        int i = 0;
-        for (String each : allDatabaseNames) {
-            openGaussDatabases[i++] = new OpenGaussDatabase(each, DAT_COMPATIBILITY);
+        List<OpenGaussTables> openGaussTables = new LinkedList<>();
+        List<OpenGaussRoles> openGaussRoles = new LinkedList<>();
+        ProxyContext.getInstance().getContextManager().getMetaDataContexts().getMetaData().getGlobalRuleMetaData().getSingleRule(AuthorityRule.class)
+                .getConfiguration().getUsers().stream().map(user -> user.getGrantee().getUsername())
+                .forEach(userName -> openGaussRoles.add(new OpenGaussRoles(userName)));
+        int index = 0;
+        for (String databaseName : allDatabaseNames) {
```
Comment (tuichenchuxin) [inline proxy/backend/type/opengauss/src/main/java/org/apache/shardingsphere/proxy/backend/opengauss/handler/admin/OpenGaussSystemCatalogAdminQueryExecutor.java:106]: better use each to replace databaseName

File: proxy/backend/type/opengauss/src/main/java/org/apache/shardingsphere/proxy/backend/opengauss/handler/admin/OpenGaussSystemCatalogAdminQueryExecutor.java [hunk 2]
```diff
@@ -88,11 +97,24 @@ private void prepareCalciteConnection(final Connection connection) throws SQLExc
...
+            for (Map.Entry<String, ShardingSphereSchema> schema : ProxyContext.getInstance().getDatabase(databaseName).getSchemas().entrySet()) {
```
Comment (tuichenchuxin) [inline proxy/backend/type/opengauss/src/main/java/org/apache/shardingsphere/proxy/backend/opengauss/handler/admin/OpenGaussSystemCatalogAdminQueryExecutor.java:107]: use `Entry` to replace `Map.Entry` 
use `entry` to replace `schema`

File: proxy/backend/type/opengauss/src/main/java/org/apache/shardingsphere/proxy/backend/opengauss/handler/admin/OpenGaussSystemCatalogAdminQueryExecutor.java [hunk 3]
```diff
@@ -88,11 +97,24 @@ private void prepareCalciteConnection(final Connection connection) throws SQLExc
...
+                String schemaName = schema.getKey();
+                for (String tableName : schema.getValue().getAllTableNames()) {
```
Comment (tuichenchuxin) [inline proxy/backend/type/opengauss/src/main/java/org/apache/shardingsphere/proxy/backend/opengauss/handler/admin/OpenGaussSystemCatalogAdminQueryExecutor.java:109]: replace tableName with each

**Review by tuichenchuxin — APPROVED** [review]

**Pull request merged**